### PR TITLE
remove boost::lexical_cast

### PR DESCRIPTION
**Pull request by ThomasHabets**
Not quite ready for merging. I need to do proper testing with the UI etc.

Feel free to do it too.

### CODE REVIEW

File: gnuradio-runtime/lib/pmt/pmt_unv.cc [hunk 1]
```diff
@@ -137,7 +138,7 @@ uint8_t* u8vector_writable_elements(pmt_t vector, size_t& len)
 
 const std::string pmt_u8vector::string_ref(size_t k) const
 {
-    return boost::lexical_cast<std::string, uint8_t>(ref(k));
+    return std::to_string(ref(k));
```
Comment (marcusmueller) [inline gnuradio-runtime/lib/pmt/pmt_unv.cc:141]: I like this!

File: gnuradio-runtime/lib/pmt/pmt_unv.cc [hunk 2]
```diff
@@ -1143,7 +1144,9 @@ float* f32vector_writable_elements(pmt_t vector, size_t& len)
 
 const std::string pmt_f32vector::string_ref(size_t k) const
 {
-    return boost::lexical_cast<std::string, float>(ref(k));
+    char buf[128];
+    snprintf(buf, sizeof(buf), "%.*f", std::numeric_limits<float>::digits10, ref(k));
```
Comment (marcusmueller) [inline gnuradio-runtime/lib/pmt/pmt_unv.cc:1148]: ha! nice, I didn't even know of `digits10`!

File: gr-qtgui/include/gnuradio/qtgui/edit_box_msg.h [hunk 3]
```diff
@@ -57,10 +57,10 @@ namespace qtgui {
  * how to handle complex numbers.
  *
  * Complex numbers are currently handled a bit differently than
- * expected. Because we use the Boost lexical_cast function,
- * complex numbers MUST be in the form "(a,b)" to represent "a +
- * jb". Note that you cannot even have a space after the comma, so
- * "(1.23,10.56)" is correct while "(1.23, 10.56)" will not parse.
+ * expected. Because we use the std iostream for complex, complex numbers MUST
```
Comment (marcusmueller) [inline gr-qtgui/include/gnuradio/qtgui/edit_box_msg.h:60]: minor typo! `std::stringstream`, not iostream :)

File: gnuradio-runtime/lib/pmt/pmt_unv.cc (hunk 1 above)
Comment (ThomasHabets) [reply gnuradio-runtime/lib/pmt/pmt_unv.cc:141]: Except it was actually a mistake. :-)

`lexical_cast` turns 0x42 into `B`, and `to_string` makes it `66`. Fixed.

**Comment by marcusmueller** [comment]
@ThomasHabets  I'd merge this right away, but it's still got your "Don't merge" tag; is that obsolete?

**Comment by ThomasHabets** [comment]
If you feel confident enough then go for it. My plan was to remove the "Don't merge" tag after running some tests in the UI, e.g. dumping some data in the message debug sink to see that it printed the same. Have not had time to do that yet.

**Comment by ThomasHabets** [comment]
Yeah I'm happy with my testing now.

**Pull request merged**